.circle/Makefile

```diff
@@ -1,249 +1,33 @@
-	if [ -z "${CHANGED_FILES}" ]; then \
-		echo "No files have changed, skipping run..."; \
-	else \
-		$(ST2_REPO_PATH)/st2common/bin/st2-run-pack-tests -c -t -x -j -p $(ROOT_DIR) || exit 1; \
-	fi;
-
-.PHONY: .packs-missing-tests
-.packs-missing-tests:
-	@echo
-	@echo "==================== pack-missing-tests ===================="
-	@echo
-	if [ -z "${CHANGED_FILES}" ]; then \
-		echo "No files have changed, skipping run..."; \
-	else \
-		st2-check-print-pack-tests-coverage $(ROOT_DIR) || exit 1; \
-	fi;
-
-.PHONY: .clone_st2_repo
-.clone_st2_repo: /tmp/st2
-/tmp/st2:
-	@echo
-	@echo "==================== cloning st2 repo ===================="
-	@echo
-	@rm -rf /tmp/st2
-	@git clone https://github.com/StackStorm/st2.git --depth 1 --single-branch --branch $(ST2_REPO_BRANCH) /tmp/st2
-
-.PHONY: .install-runners
-.install-runners:
-	@echo ""
-	@echo "================== install runners ===================="
-	@echo ""
-	@for component in $(COMPONENTS_RUNNERS); do \
-		echo "==========================================================="; \
-		echo "Installing runner:" $$component; \
-		echo "==========================================================="; \
-        (. $(VIRTUALENV_DIR)/bin/activate; cd $$component; python setup.py develop); \
-	done
-	@echo ""
-	@echo "================== register metrics drivers ======================"
-	@echo ""
-
-	# Install st2common to register metrics drivers
-	(. $(VIRTUALENV_DIR)/bin/activate; cd $(ST2_REPO_PATH)/st2common; python setup.py develop)
-
-.PHONY: requirements
-requirements: virtualenv .clone_st2_repo .install-runners
-	@echo
-	@echo "==================== requirements ===================="
-	@echo
-	. $(VIRTUALENV_DIR)/bin/activate && $(VIRTUALENV_DIR)/bin/pip install --upgrade "pip>=9.0,<9.1"
-	. $(VIRTUALENV_DIR)/bin/activate && $(VIRTUALENV_DIR)/bin/pip install --cache-dir $(HOME)/.pip-cache -q -r $(CI_DIR)/.circle/requirements-dev.txt
-	. $(VIRTUALENV_DIR)/bin/activate && $(VIRTUALENV_DIR)/bin/pip install --cache-dir $(HOME)/.pip-cache -q -r $(CI_DIR)/.circle/requirements-pack-tests.txt
-
-.PHONY: requirements-ci
-requirements-ci:
-	@echo
-	@echo "==================== requirements-ci ===================="
-	@echo
-	. $(VIRTUALENV_DIR)/bin/activate && $(VIRTUALENV_DIR)/bin/pip install --upgrade "pip>=9.0,<9.1"
-	. $(VIRTUALENV_DIR)/bin/activate && $(VIRTUALENV_DIR)/bin/pip install --cache-dir $(HOME)/.pip-cache -q -r $(CI_DIR)/.circle/requirements-dev.txt
-	. $(VIRTUALENV_DIR)/bin/activate && $(VIRTUALENV_DIR)/bin/pip install --cache-dir $(HOME)/.pip-cache -q -r $(CI_DIR)/.circle/requirements-pack-tests.txt
-
-.PHONY: virtualenv
 virtualenv: $(VIRTUALENV_DIR)/bin/activate
 $(VIRTUALENV_DIR)/bin/activate:
 	@echo
 	@echo "==================== virtualenv ===================="
 	@echo
 	test -d $(VIRTUALENV_DIR) || virtualenv --no-site-packages $(VIRTUALENV_DIR)
+
+$(VIRTUALENV_DIR)/bin/invoke: $(VIRTUALENV_DIR)
+	. $(VIRTUALENV_DIR)/bin/activate && pip install invoke
```

Comment:
Do we want to pin a version?

Reply:
I don't think so, since that would immediately become a maintenance burden. Invoke was born out of the Fabric project, so the developers have had a good amount of time to figure out what works and what doesn't. I think that means that the API is pretty stable and won't really change a whole lot from here on out. And we aren't really using any in-depth parts of Invoke's API, so our code is very likely to remain compatible with newer versions of Invoke.